test/functional/p2p_ibd_txrelay.py

```diff
@@ -28,6 +47,31 @@ def run_test(self):
             assert node.getblockchaininfo()['initialblockdownload']
             self.wait_until(lambda: all(peer['minfeefilter'] == MAX_FEE_FILTER for peer in node.getpeerinfo()))
 
+        self.log.info("Check that nodes don't send getdatas for transactions while still in IBD")
+        peer_inver = self.nodes[0].add_p2p_connection(P2PDataStore())
+        txid = 0xdeadbeef
+        peer_inver.send_and_ping(msg_inv([CInv(t=MSG_WTX, h=txid)]))
+        # The node should not send a getdata, but if it did, it would first delay 2 seconds
+        self.nodes[0].setmocktime(int(time.time() + NONPREF_PEER_TX_DELAY))
+        peer_inver.sync_with_ping()
```

Comment:
```suggestion
        peer_inver.sync_send_with_ping()
```

To account for the possibility that pongs are sent before getdata?

Reply:
Yes :brain:

Reply:
Wait, isn't there a ping after the inv? since we're calling `send_and_ping()`?

Reply:
There is, but it doesn't matter. The pong in this line (after mocktime) will be sent out here:

```
            // Receive messages
            bool fMoreNodeWork = m_msgproc->ProcessMessages(pnode, flagInterruptMsgProc);
```

At this point no messages have been sent out (no getdata etc). Sending will happen right after, but may take a long time if the thread is put aside or the buffer is full.

Reply:
Rebased for #21785 and added this